rai/node/rpc.cpp

```diff
@@ -1809,6 +1809,8 @@ void rai::rpc_handler::account_history ()
 				boost::property_tree::ptree history;
 				if (!error)
 				{
+					auto account (node.ledger.account (transaction, hash));
```

Comment:
This is a really expensive operation. Seems like you can just use the input account for account queries, and only do the expensive lookup for head queries?

Reply:
It's not expensive if the supplied block is a head block though, right?

Reply:
It looks through up to 5 tables for the block (block_get_raw), then does a block_info_get. All that instead of just setting the output string to the input string.

Reply:
Fair enough.

Reply:
The `history_visitor` does this call for every single open, receive and state block already, so the overhead would be negligible for history calls with a somewhat high `count`. I do agree this is a reasonable optimization, though.

My rationale for doing the lookup was because I assumed the passed parameter was case insensitive and therefore this would double as a normalization of the account string, but I see now that this is not the case.

Reply:
Theoretically the account parameter could be encoded using the old v1 account format, but realistically whatever. I'm not too concerned about if we optimize this or not.